packages/discord.js/src/structures/ThreadChannel.js

```diff
@@ -101,7 +101,19 @@ class ThreadChannel extends Channel {
        * @type {?number}
        */
       this.archiveTimestamp = Date.parse(data.thread_metadata.archive_timestamp);
+
+      if ('create_timestamp' in data.thread_metadata) {
+        /**
+         * The timestamp when this thread was created. This isn't available for threads
+         * created before 2022-01-09
+         * @type {?number}
+         */
+        this.createdTimestamp = Date.parse(data.thread_metadata.create_timestamp);
+      } else {
+        this.createdTimestamp ??= null;
```

Comment:
Okay this is gonna be slightly painful, but on private threads you are guaranteed the id of the thread is the created timestamp (can't create a private thread from an existing message), so if createdTimestamp is nullish and not present in the payload I think we can get away doing  `this.createdTimestamp ??= this.type === ChannelTypes.PrivateThread ? super.createdTimestamp :  null`